pkg/restapi/vc/operation/operations.go

```diff
@@ -773,32 +774,46 @@ func (o *Operation) writeErrorResponse(rw http.ResponseWriter, status int, msg s
 }
 
 func (o *Operation) issueCredentialHandler(rw http.ResponseWriter, req *http.Request) {
+	// get the issuer profile
+	profileID := mux.Vars(req)[profileIDPathParam]
+
+	profile, err := o.profileStore.GetProfile(profileID)
+	if err != nil {
+		o.writeErrorResponse(rw, http.StatusBadRequest, fmt.Sprintf("invalid issuer profile - id=%s: err=%s",
+			profileID, err.Error()))
+
+		return
+	}
+
 	// get the request
 	cred := IssueCredentialRequest{}
 
-	err := json.NewDecoder(req.Body).Decode(&cred)
+	err = json.NewDecoder(req.Body).Decode(&cred)
 	if err != nil {
 		o.writeErrorResponse(rw, http.StatusBadRequest, fmt.Sprintf(invalidRequestErrMsg+": %s", err.Error()))
 
 		return
 	}
 
 	// validate the VC
-	validatedCred, _, err := verifiable.NewCredential(cred.Credential)
+	credential, _, err := verifiable.NewCredential(cred.Credential)
 	if err != nil {
 		o.writeErrorResponse(rw, http.StatusBadRequest, fmt.Sprintf("failed to validate credential: %s", err.Error()))
 
 		return
 	}
 
-	// use issuer id if options doesn't specify the DID
-	signerDID := validatedCred.Issuer.ID
-	if cred.Opts != nil && cred.Opts.AssertionMethod != "" {
-		signerDID = cred.Opts.AssertionMethod
+	// update the signing profile with the request options
+	err = updateIssueCredSigningProfile(o.vdri, &cred, profile)
+	if err != nil {
+		o.writeErrorResponse(rw, http.StatusBadRequest, fmt.Sprintf("failed to update signing profile:"+
```

Comment:
http error code should be internal server error,

Reply:
In this case, we are updating the profile based on request (&cred is a request). Hence added bad-request.

Reply:
I saw call to vdri there, that's why.

Reply:
yeah but we are using VDRI to resolve user provided DID.